src/taker/offers.rs

```diff
@@ -154,11 +190,86 @@ pub async fn sync_offerbook_with_addresses(
 
 /// Retrieves advertised maker addresses from directory servers based on the specified network.
 pub async fn get_advertised_maker_addresses(
-    network: Network,
+    socks_port: Option<u16>,
+    directory_server_address: Option<String>,
+    _network: Network,
 ) -> Result<Vec<MakerAddress>, DirectoryServerError> {
-    Ok(if network == Network::Regtest {
-        get_regtest_maker_addresses()
-    } else {
-        sync_maker_addresses_from_directory_servers(network).await?
-    })
+    let mut directory_onion_address = match directory_server_address {
+        Some(address) => address,
+        None => "".to_string(),
+    };
+    if cfg!(feature = "integration-test") {
+        let directory_hs_path_str = "/tmp/tor-rust-directory/hs-dir/hostname".to_string();
+        let directory_hs_path = PathBuf::from(directory_hs_path_str);
+        let mut directory_file = tokio::fs::File::open(&directory_hs_path)
+            .await
+            .map_err(|_e| DirectoryServerError::Other("Directory hidden service path not found"))?;
+        let mut directory_onion_addr = String::new();
+        directory_file
+            .read_to_string(&mut directory_onion_addr)
+            .await
+            .map_err(|_e| DirectoryServerError::Other("Reading onion address failed"))?;
+        directory_onion_addr.pop();
+        directory_onion_address = format!("{}:{}", directory_onion_addr, 8080);
+    }
```

Comment:
Lets put this logic to the parent function. `sync_offerbook`. Directory address setting logic isn't a contextual fit here.

We do the same thing we did for makers. `cfg!` checks to find the address from local dir in IT or else get it from the config values. And should be handled at sync_offerbook.

Reply:
makes sense